app/page.tsx

```diff
@@ -1,47 +1,77 @@
-import { GetServerSideProps } from 'next'
-import Image from 'next/image'
+import HeatMap from './calendar-heatmap';
 
-async function getCalendar(): Promise<Calendar> {
-  const res = await fetch('http://127.0.0.1:8080/calendar/id')
+export const dynamic = 'force-dynamic'
 
-  if(!res.ok){
-    throw new Error('Failed to fetch data')
+export default async function Page() {
+  let now = new Date();
+
+  let startDate = new Date(now);
+  startDate.setMonth(startDate.getMonth() - 6)
+  
+  let endDate = new Date(now);
+  endDate.setMonth(endDate.getMonth() + 6)
+
+  const calendar = await getCalendar()
+  console.log(calendar)
+  for(var event of calendar.events) {
+    event.date_time = new Date(event.date_time);
   }
 
-  return res.json()
-}
+  let values = [];
+  for (var d = new Date(startDate); d <= endDate; d.setDate(d.getDate() + 1)) {
```

Comment:
Couldn't this for in for be improved?

Reply:
Yes, you are right, we can do better.
I added a new task on our todo list, for now the current version is good enough